CMakeLists.txt

```diff
@@ -4,17 +4,17 @@ project(swipl-cpp)
 include("../cmake/PrologPackage.cmake")
 
 if(MSVC)
-set(CMAKE_CXX_STANDARD 20) # Some users of this require C++-17
+set(CMAKE_CXX_STANDARD 20)
 else()
-set(CMAKE_CXX_STANDARD 17) # Some users of this require C++-17
+set(CMAKE_CXX_STANDARD 17) # C++-17 ought to suffice, C++-20 is better?
 endif()
-set(CMAKE_C_STANDARD 11) # swipl requires C-11
+set(CMAKE_C_STANDARD 17) # swipl requires C-11
 
 configure_file(config.h.cmake config.h)
 
 install_src(pkg_cpp_headers
-	    FILES SWI-cpp.h SWI-cpp2.h SWI-cpp2.cpp SWI-cpp2-plx.h SWI-cpp2-atommap.h DESTINATION
-	    ${SWIPL_INSTALL_INCLUDE})
+	    FILES SWI-cpp.h SWI-cpp2.h SWI-cpp2.cpp SWI-cpp2-plx.h SWI-cpp2-atommap.h SWI-cpp2-flags.h
```

Comment:
Do we really want another public header for this?   It seems to do reflection on bitmasks to make the test file a little simpler.    I do not see that many practical use cases and I'd consider that more some generic problem rather than something to solve in SWI-Prolog.    Of course, the header is fine, but why nost just keep it in the source tree?